src/mechanics/timeIntegration/ExplicitRungeKutta.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <vector>
+
+namespace NuTo
+{
+namespace TimeIntegration
+{
+template <typename state>
+class ExplicitRungeKutta
+{
+
+public:
+    //! @brief Initialization with method specific parameters (butcher tableau)
+    ExplicitRungeKutta(std::vector<std::vector<double>> aa, std::vector<double> bb, std::vector<double> cc)
+    {
+        for (auto elm : aa)
+        {
+            a.push_back(elm);
+        }
+        for (auto elm : bb)
+        {
+            b.push_back(elm);
+        }
+        for (auto elm : cc)
+        {
+            c.push_back(elm);
+        }
+    }
+
+    //! @brief Performs one RungeKutta step
+    template <typename F>
+    state doStep(F f, state w0, double t0, double h)
+    {
+        std::vector<state> k;
+        for (size_t i = 0; i < c.size(); i++)
+        {
+            k.push_back(state(w0));
```

Comment:
Is it really necessary to dynamically allocate  the state vectors for each time step?  It would be interesting to see the performance difference, if this allocation and the state vector would be moved to the constructor/member variable , the number of variables should be known there.

Reply:
The size of the state vector is not known when the integrator is created. But of course once the integration starts things are different. This will probably only be relevant for large state vectors and I want to postpone optimizations of this kind to later stages. Maybe a function like integrateNsteps() could take care of the initial resizing.